script/packagecloud.rb

```diff
@@ -8,9 +8,13 @@
 
 require "json"
 
+packagecloud_ruby_minimum_version = "1.0.4"
 begin
+  gem "packagecloud-ruby", ">=#{packagecloud_ruby_minimum_version}"
   require "packagecloud"
```

Comment:
Thanks @ttaylorr for the feedback.  I am struggling a bit to apply it though.

Did you mean to move, rather than copy the `require "packagecloud"` statement, to be outside of the `rescue` block?  I myself found the ruby mechanism for forcing the specific version or version minimum confusing here, but I have intentionally left both `gem` and `require` statements inside the `begin/rescue/end` block.  The purpose of this block is to inform the developer that they don't have the required package/version, and fail-fast, preventing a deployment attempt.  My understanding is that you cannot enforce the version with `require`, and `gem` will test that, but the `require` is still required.

So, I don't know whether `gem` could pass but `require` fail - this might be what you are angling at - that the `require` no longer needs to be in the block.  I don't mind either way really - it looks OK to me now, but I'm no Ruby expert.